src/eq.cpp

```diff
@@ -50,9 +50,25 @@ void *MainFrame::designAnEQFilter(const char filterType[], float freqHz, float g
 
 void  MainFrame::designEQFilters(paCallBackData *cb, int rxSampleRate, int txSampleRate)
 {
+    bool deleteFilters = false;
+    
+    if (m_newMicInFilter && (cb->sbqMicInBass || cb->sbqMicInTreble || cb->sbqMicInMid))
+    {
```

Comment:
Won't cb->sbqMicInBass, cb->sbqMicInTreble and cb->sbqMicInMid, always be non-null?  Is it possible to have a condition where just one of them is null?

Reply:
Thinking about it more, I don't think there is. I'll assert, though, as that should probably be fixed if someone does run into such a case.

Reply:
I agree - an assert is a good idea :+1:  Although this code has been pretty stable for many years, so I'd be surprising if it has any problems.  Still, multi-rx might introduce new conditions.

Does this mean these changes to `eq.cpp` are not necessary and can be reverted?

Reply:
Yep, will do.

Reply:
So can we also revert the changes in `designEQFilters()` between line 55 and 75 :point_up: :point_down: ie is there a specific need for all the if then clauses checking each filter?

Reply:
The latest commit only checks sbqMicInBass and sbqSpkOutBass != nullptr and asserts on the others. For example:

```
    if (m_newMicInFilter && cb->sbqMicInBass != nullptr)
    {
        // All three should be valid if one is, so we assert that here.
        assert(cb->sbqMicInBass != nullptr && cb->sbqMicInTreble != nullptr && cb->sbqMicInMid != nullptr);
        
        deleteFilters = true;
    }
```

Reply:
Gotcha. So why is it necessary to check `cb->sbqMicInBass != nullptr` ? Sorry if I'm  missing something - I'm unsure what problem the lines 73-75 actually fix?

Reply:
deleteEQFilters() still asserts on the filters being not null prior to calling sox_biquad_delete(). Even if it didn't, sox_biquad_delete() dereferences the pointers passed into it to free the filters' internal data prior to setting them to nullptr. The setting to nullptr is to ensure that we don't inadvertently touch already-freed memory (in other words, fail much more quickly if it somehow did touch the filters when they were invalid).

Reply:
Plus, failing immediately on attempted dereference means that it should be easier to figure out what happened vs. having potentially inconsistent behavior by touching memory it shouldn't.

Reply:
> deleteEQFilters() still asserts on the filters being not null prior to calling sox_biquad_delete(). 

OK, good catch.  So the core issue is `deleteEQFilters() ` and `designEQFilters` are being called out of order? I agree with the asserts and assignment of pointers to null to highlight the problem.  But I feel we should track down the root cause for asserts firing.  I'm happy to look into this one myself if you can tell me how to reproduce the problem.

Reply:
I'm not sure there's actually a problem at the moment now. The main issue was the memory leaks around the filters, and I think that was solved by the change in sox_biquad.c lines 95 and 104 + setting m_newMicInFilter and m_newSpkOutFilter to true prior to deleteEQFilters() in stopRxStream().